onnxruntime/core/providers/webnn/builders/impl/triangular_op_builder.cc

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Copyright (c) Intel Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/common.h"
+#include "core/providers/shared/utils/utils.h"
+#include "core/providers/webnn/builders/helper.h"
+#include "core/providers/webnn/builders/model_builder.h"
+#include "core/providers/webnn/builders/op_builder_factory.h"
+
+#include "base_op_builder.h"
+
+namespace onnxruntime {
+namespace webnn {
+
+class TriangularOpBuilder : public BaseOpBuilder {
+  // Add operator related.
+ public:
+  void AddInitializersToSkip(ModelBuilder& model_builder, const Node& node) const override;
+
+ private:
+  Status AddToModelBuilderImpl(ModelBuilder& model_builder, const Node& node,
+                               const logging::Logger& logger) const override ORT_MUST_USE_RESULT;
+
+  // Operator support related.
+ private:
+  bool IsOpSupportedImpl(const InitializedTensorSet& initializers, const Node& node,
+                         const WebnnDeviceType /* device_type */, const logging::Logger& logger) const override;
+};
+
+// Add operator related.
+
+void TriangularOpBuilder::AddInitializersToSkip(ModelBuilder& model_builder, const Node& node) const {
+  // Skip diagonal initializer if present.
+  if (node.InputDefs().size() > 1) {
+    model_builder.AddInitializerToSkip(node.InputDefs()[1]->Name());
+  }
+}
+
+Status TriangularOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder,
+                                                  const Node& node,
+                                                  const logging::Logger& logger) const {
+  const auto& input_defs = node.InputDefs();
+  const auto& initializers = model_builder.GetInitializerTensors();
+  emscripten::val input = model_builder.GetOperand(input_defs[0]->Name());
+  emscripten::val output = emscripten::val::object();
+  NodeAttrHelper helper(node);
+  emscripten::val options = emscripten::val::object();
+
+  const bool upper = helper.Get("upper", 1);
+  options.set("upper", upper);
+
+  if (!GetTensorName(input_defs, 1).empty()) {
+    // Optional input diagonal is provided, use diagonal initializer data.
+    const auto diagonal_tensor = *initializers.at(input_defs[1]->Name());
```

Comment:
(minor) On line 85 you get the name and then reuse it. Could have done that here too rather than query the name twice.